ExpenseMaster/Middlewares/JwtTokenMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Extensions.Primitives;
+using Microsoft.IdentityModel.Tokens;
+using System.IdentityModel.Tokens.Jwt;
+using System.Text;
+
+namespace ExpenseMaster.Middlewares
+{
+    public class JwtTokenMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IConfiguration _configuration;
+
+        public JwtTokenMiddleware(RequestDelegate next, IConfiguration configuration)
+        {
+            _next = next;
+            _configuration = configuration;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            if (!context.Request.Headers.TryGetValue("Authorization", out StringValues authHeader))
+            {
+                context.Response.StatusCode = 401;
+                await context.Response.WriteAsync("Authorization header not found");
+
+                return;
+            }
+
+            if (!authHeader.ToString().StartsWith("Bearer "))
+            {
+                context.Response.StatusCode = 401;
+                await context.Response.WriteAsync("Invalid authorization header format");
+
+                return;
+            }
+            
+            string token = authHeader.ToString().Substring(7);
+
+            try
+            {
+                var tokenHandler = new JwtSecurityTokenHandler();
```

Comment:
You can write these configs, when you register in di(like AddAuthentification(opt => ....))